fdbserver/workloads/MetaclusterManagementWorkload.actor.cpp

```diff
@@ -392,6 +392,39 @@ struct MetaclusterManagementWorkload : TestWorkload {
 		return Void();
 	}
 
+	ACTOR static Future<Void> verifyListFilter(MetaclusterManagementWorkload* self, TenantName tenant) {
+		try {
+			state TenantMapEntry checkEntry = wait(MetaclusterAPI::getTenant(self->managementDb, tenant));
+			state TenantState checkState = checkEntry.tenantState;
+			state std::vector<TenantState> filters;
+			filters.push_back(checkState);
+			state std::vector<std::pair<TenantName, TenantMapEntry>> tenantList;
+			// Possible to have changed state between now and the getTenant call above
+			state TenantMapEntry checkEntry2;
+			wait(store(checkEntry2, MetaclusterAPI::getTenant(self->managementDb, tenant)) &&
+			     store(tenantList,
+			           MetaclusterAPI::listTenants(self->managementDb, ""_sr, "\xff\xff"_sr, 10e6, 0, filters)));
+			if (tenantList.empty()) {
+				ASSERT(checkEntry2.tenantState != checkState);
+				return Void();
+			}
+			bool found = false;
+			for (auto pair : tenantList) {
+				ASSERT(pair.second.tenantState == checkState);
+				if (pair.first == tenant) {
+					found = true;
+				}
+			}
+			ASSERT(found || checkEntry2.tenantState != checkState);
```

Comment:
With this line here, I think we could get rid of the if block at line 407 since it is just a special case of the check here.